services/user/XAdmin/ui/src/lib/bootChain.ts

```diff
@@ -25,11 +28,26 @@ export const bootChain = async (
             packages.map((pack) => pack.file)
         );
 
+        let publicKeyPem: string | undefined;
+        try {
+            if (publicKey) {
+                publicKeyPem = await exportKeyToPEM(publicKey, "PUBLIC KEY");
+            }
+        } catch (e) {
+            onProgressUpdate("Failed to export publicKey to PEM format");
+            return;
+        }
+
         // Something is wrong with the Vite proxy configuration that causes boot to intermittently (but often) fail
         // in a dev environment.
 
         const [boot_transaction, transactions] =
-            wasm.js_create_boot_transactions(producerName, fetchedPackages);
+            wasm.js_create_boot_transactions(
+                producerName,
+                fetchedPackages,
+                publicKeyPem,
+                4
```

Comment:
I think I mentioned you could just hardcode the compression as 4, but now that I think about it, it's probably better to use 11 for the production template, 4 for the development template.

Reply:
Okay. I'll address this.

Reply:
Are we going with 11 because of the presence of the public key, or specifically because it's Production? Not sure what "compression" is here.

Reply:
7 seemed to strike the right balance with UI responsiveness. FYI.